resources/aws/launch_configuration.go

```diff
@@ -0,0 +1,75 @@
+package aws
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/autoscaling"
+	microerror "github.com/giantswarm/microkit/error"
+)
+
+// LaunchConfiguration is a template for launching EC2 instances into an auto
+// scaling group.
+type LaunchConfiguration struct {
+	Name                   string
+	IamInstanceProfileName string
+	ImageID                string
+	InstanceType           string
+	KeyName                string
+	SecurityGroupID        string
+	SmallCloudConfig       string
+
+	// Dependencies
+	Client *autoscaling.AutoScaling
+}
+
+// CreateIfNotExists creates the launch config if it doesn't exist.
+func (lc *LaunchConfiguration) CreateIfNotExists() (bool, error) {
+	if lc.Client == nil {
+		return false, microerror.MaskAny(clientNotInitializedError)
+	}
+
+	if err := lc.CreateOrFail(); err != nil {
```

Comment:
Here we should check if the `CreateOrFail` returned a `ErrCodeAlreadyExistsFault`, and in that case return `false, nil`.